### PR TITLE
Adds GraphQL API for adding books

### LINKED ISSUE #1: Add a GraphQL API for adding Books
Neither a persistence layer nor authentication is needed yet, but the basic structure so the front can start working.

### PR DESCRIPTION
**Pull request by disalazarg**
Resolves #1 

Uses an alpha version of https://bitbucket.org/disalazarg/reflex for persistence for now.

### CODE REVIEW

**Comment by disalazarg** [comment]
Force-pushed a `git rebase` to get CI checks on this branch

File: apps/shelver/mix.exs [hunk 1]
```diff
@@ -36,6 +36,7 @@ defmodule Shelver.MixProject do
   # Type `mix help deps` for examples and options.
   defp deps do
     [
+      {:reflex, git: "https://bitbucket.org/disalazarg/reflex.git"},
```
Comment (sespinoza-dev) [inline apps/shelver/mix.exs:40]: are you using a library that you created?

**Review by sespinoza-dev — COMMENTED** [review]
looks nice, I only have a doubt on the reference to a library in bitbucket.

File: apps/shelver/mix.exs (hunk 1 above)
Comment (disalazarg) [reply apps/shelver/mix.exs:40]: @sespinoza-dev yup, available publicly albeit licenseless as sort of a pre-release alpha, this project serving as its first test in the real world; I'm planning to refactor it to integrate more cleanly with Absinthe/GraphQL, give it a F/OSS license and publish it on hex afterwards. Hopefully, before next year.

**Pull request merged**